src/etherscan/verifyProxyContract.ts

```diff
@@ -0,0 +1,95 @@
+import axios from "axios";
+import { config } from "../config";
+import { partyLogicAddresses } from "./constants";
+import { PartyEvent } from "types";
+import {
+  getEtherscanVerifyContractGuid,
+  getEtherscanVerifyProxyContractGuid,
+  setEtherscanVerifyProxyContractGuid,
+} from "./helpers";
+
+/**
+ * Verify a proxy contract using the etherscan api
+ * etherscan docs - https://docs.etherscan.io/api-endpoints/contracts#verify-proxy-contract
+ * @param {string} address - party address
+ * @returns {{ isSuccess: boolean, guid?: string}} - true if verified, false otherwise
+ */
+export const verifyProxyContract = async (
+  address: string
+  expectedImplementationAddress?: string
+): Promise<{ isSuccess: boolean; guid?: string }> => {
+  if (!address) {
+    throw new Error("Missing argument: address");
+  }
+
+  try {
+    const params = new URLSearchParams();
+    params.append("address", address);
+    // note: if expectedImplementationAddress is not provided, etherscan will try to determine the implementation
+    // based on similar contracts
+    if (expectedImplementationAddress) {
+      params.append("expectedimplementation", expectedImplementationAddress);
+    }
+    const resp = await axios.post(
+      `${config.etherscan.apiBase}/?module=contract&action=verifyproxycontract&apikey=${config.etherscan.apiKey}`,
+      params
+    );
+
+    if (!(resp && resp.data)) {
+      throw new Error(`Invalid response from Etherscan API for ${address}`);
+    }
+    const { data } = resp;
+    console.log("etherscan:: data", data);
+    if (data.status === "1" && data.result) {
+      const guid = data.result;
+      console.log(
+        `etherscan:: successfully verified proxy contract for ${address}. guid ${guid}.`
+      );
+      return { isSuccess: true, guid };
+    } else {
+      throw new Error(`Could not verify proxy contract for ${address}`);
+    }
+  } catch (error) {
+    console.error(error);
+    return { isSuccess: false };
+  }
+};
+
+/**
+ * verify proxy contract on etherscan if the event is a start event
+ * @param {PartyEvent} event
+ * @returns true/false if verified
+ */
+export const verifyProxyContractForEvent = async (
+  event: PartyEvent
+): Promise<boolean> => {
+  // todo: check etherscan API to see if collection party source code is verified
```

Comment:
can take out this todo

Reply:
i left that comment in because we're technically not checking the etherscan api for checking if the contract was successfully verified. instead, we are checking to see if we have a guid from calling the verify contract api.

Reply:
got it 👍